### PR TITLE
Profile backend

### PR DESCRIPTION
**Pull request by JasonTuTW**
改了user api的路徑和加加改改competition/user info
(恩...改的部分有點多
image

### CODE REVIEW

File: backend/config/db.yaml
Comment (lolainta) [inline backend/config/db.yaml:1]: Leave a `db.yaml.sample` here and use git ignore to ignore the real credentials

File: backend/internal/model/init.go [hunk 1]
```diff
@@ -25,15 +25,17 @@ type Competition struct {
...
+type CompetitionCategory struct {
+	CompetitionID 	uint `gorm:"not null"`
+	Description 	string  `gorm:"not null"`
+	Distance 		int32 `gorm:"not null"`
 }
 
 type Participant struct {
-	UserID uint `gorm:"primaryKey"`
-	CompetitionID uint `gorm:"not null"`
+	UserID 			uint `gorm:"not null"`
```
Comment (lolainta) [inline backend/internal/model/init.go:37]: Why not primary key?

File: backend/main.go [hunk 2]
```diff
@@ -25,18 +25,22 @@ func routerSetup() *gin.Engine {
 		sr.GET("/", pkg.AuthSessionMiddleware(), func(c *gin.Context) {
 			c.JSON(http.StatusOK, gin.H{"result": "ok"});
 		})
+		
+		userssr := sr.Group("/user")
+		{
+			userssr.POST("/register", api.Register)
+			userssr.POST("/login", api.Login)
+			userssr.GET("/logout", api.Logout)
+			userssr.POST("/modifyInfo", pkg.AuthSessionMiddleware(), api.ModifyInfo)
```
Comment (lolainta) [inline backend/main.go]: It is better to use PUT for update only

File: backend/main.go [hunk 3]
```diff
@@ -25,18 +25,22 @@ func routerSetup() *gin.Engine {
...
+			userssr.GET("/getUserID", pkg.AuthSessionMiddleware(), api.GetUserID)
+			userssr.GET("/info/:id", api.UserInfo)
+		}
 
-		sr.POST("/register", api.Register)
-		sr.POST("/login", api.Login)
-		sr.GET("/logout", api.Logout)
-		sr.POST("/modifyInfo", pkg.AuthSessionMiddleware(), api.ModifyInfo)
```
Comment (lolainta) [inline backend/main.go:32]: dito

File: backend/main.go [hunk 4]
```diff
@@ -25,18 +25,22 @@ func routerSetup() *gin.Engine {
...
-		sr.GET("/getUsername", pkg.AuthSessionMiddleware(), api.GetUsername)
-		sr.POST("userInfo", api.UserInfo)
-
-		ssr := sr.Group("/competition", pkg.AuthSessionMiddleware())
+		compssr := sr.Group("/competition")
 		{
-			ssr.POST("/create", api.CreateCompetition)
-			ssr.POST("/join", api.JoinInCompetition)
+			compssr.POST("/create", api.CreateCompetition, pkg.AuthSessionMiddleware())
+			compssr.POST("/join", api.JoinInCompetition, pkg.AuthSessionMiddleware())
```
Comment (lolainta) [inline backend/main.go:42]: dito

File: backend/api/user.go [hunk 5]
```diff
@@ -6,6 +6,7 @@ import (
 	//"fmt"
 	"backend/internal/pkg"
 	"backend/internal/model"
+	"strconv"
```
Comment (lolainta) [inline backend/api/user.go:9]: where did you use it?

File: backend/internal/model/init.go (hunk 1 above)
Comment (JasonTuTW) [reply backend/internal/model/init.go:37]: A user can join in multiple competitions, and it seems that GORM assumes the primaryKey is unique(?)

Comment (lolainta) [reply backend/internal/model/init.go:37]: Make sense

File: backend/api/user.go (hunk 5 above)
Comment (JasonTuTW) [reply backend/api/user.go:9]: Line 133

File: backend/config/db.yaml
Comment (JasonTuTW) [reply backend/config/db.yaml:1]: Is pushing a sample "db.yaml" and adding .gitignore a better way?

Comment (lolainta) [reply backend/config/db.yaml:1]: It's alright

Comment (lolainta) [reply backend/config/db.yaml:1]: But git already tracked this file. It seems that we can't upload it without ignoring it.

**Comment by lolainta** [comment]
I think it's not possible to upload an untracked file.

File: backend/config/db.yaml
Comment (JasonTuTW) [reply backend/config/db.yaml:1]: should be fixed...

File: backend/main.go (hunk 2 above)
Comment (JasonTuTW) [reply backend/main.go]: done

**Pull request merged**